src/Aspire.Hosting.Azure/Provisioning/Internal/RunModeProvisioningContextProvider.cs

```diff
@@ -0,0 +1,145 @@
+    ITokenCredentialProvider tokenCredentialProvider,
+    DistributedApplicationExecutionContext distributedApplicationExecutionContext,
+    IOptions<PublishingOptions> publishingOptions) : BaseProvisioningContextProvider(
+        interactionService,
+        options,
+        environment,
+        logger,
+        armClientProvider,
+        userPrincipalProvider,
+        tokenCredentialProvider,
+        distributedApplicationExecutionContext,
+        publishingOptions)
+{
+    protected override string GetDefaultResourceGroupName()
+    {
+        var prefix = "rg-aspire";
+
+        if (!string.IsNullOrWhiteSpace(_options.ResourceGroupPrefix))
+        {
+            prefix = _options.ResourceGroupPrefix;
+        }
+
+        var suffix = RandomNumberGenerator.GetHexString(8, lowercase: true);
+
+        var maxApplicationNameSize = ResourceGroupNameHelpers.MaxResourceGroupNameLength - prefix.Length - suffix.Length - 2; // extra '-'s
+
+        var normalizedApplicationName = ResourceGroupNameHelpers.NormalizeResourceGroupName(_environment.ApplicationName.ToLowerInvariant());
+        if (normalizedApplicationName.Length > maxApplicationNameSize)
+        {
+            normalizedApplicationName = normalizedApplicationName[..maxApplicationNameSize];
+        }
+
+        // Run mode always includes random suffix for uniqueness
+        return $"{prefix}-{normalizedApplicationName}-{suffix}";
+    }
+
+    protected override async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, CancellationToken cancellationToken = default)
+    {
+        var locations = typeof(AzureLocation).GetProperties(BindingFlags.Public | BindingFlags.Static)
+                            .Where(p => p.PropertyType == typeof(AzureLocation))
+                            .Select(p => (AzureLocation)p.GetValue(null)!)
+                            .Select(location => KeyValuePair.Create(location.Name, location.DisplayName ?? location.Name))
+                            .OrderBy(kvp => kvp.Value)
+                            .ToList();
+
+        while (_options.Location == null || _options.SubscriptionId == null)
+        {
+            var messageBarResult = await _interactionService.PromptNotificationAsync(
+                 AzureProvisioningStrings.NotificationTitle,
+                 AzureProvisioningStrings.NotificationMessage,
+                 new NotificationInteractionOptions
+                 {
+                     Intent = MessageIntent.Warning,
+                     PrimaryButtonText = AzureProvisioningStrings.NotificationPrimaryButtonText
+                 },
+                 cancellationToken)
+                 .ConfigureAwait(false);
+
+            if (messageBarResult.Canceled)
+            {
+                // User canceled the prompt, so we exit the loop
+                _provisioningOptionsAvailable.SetException(new MissingConfigurationException("Azure provisioning options were not provided."));
```

Comment:
Was this always there?

Reply:
It was added in https://github.com/dotnet/aspire/commit/ab40bf0cf6d51e7a95501e44e73b5ceaed53f7a2#diff-b786aea8854e69ecd653511e6df511dc6dc004c8f939d5a5ad083b72b9220005R89 to exit out of the loop since the user dismissed the "The model contains Azure resources that require an Azure Subscription." message bar.